portal/account/account.lib.php

```diff
@@ -152,42 +151,51 @@ function doCredentials($pid)
     global $srcdir;
     require_once("$srcdir/authentication/common_operations.php");
 
-    $newpd = sqlQuery("SELECT * FROM `patient_data` WHERE `pid`=?", array(
+    $newpd = sqlQuery("SELECT id,fname,mname,lname,email,email_direct FROM `patient_data` WHERE `pid`=?", array(
         $pid
     ));
 
-    $clear_pass = generatePassword();
-
+    $crypto = new CryptoGen();
     $uname = $newpd['fname'] . $newpd['id'];
+    // Token expiry 1 hour
+    $expiry = new DateTime('NOW');
+    $expiry->add(new DateInterval('PT01H'));
 
-    $res = sqlStatement("SELECT * FROM patient_access_onsite WHERE pid=?", array(
-        $pid
-    ));
-    $query_parameters = array(
-        $uname
-    );
-    $salt_clause = "";
-    // For onsite portal create a blowfish based hash and salt.
+    $clear_pass = generatePassword();
+    $token_new = RandomGenUtils::createUniqueToken(32);
+    $pin = RandomGenUtils::createUniqueToken(6);
+
+    $token = $crypto->encryptStandard($token_new, '', 'database');
```

Comment:
for me:
use key from drive and do check to standard check to ensure token not empty/false